lib/minitest/ci_plugin.rb

```diff
@@ -20,4 +21,151 @@ def self.plugin_ci_options opts, options
     end
   end
 
+  def self.plugin_ci_init options
+    if Ci.report?
+      self.reporter << Ci.new(options[:io], options)
+    end
+  end
+
+  class Ci < Reporter
+    class << self
+      ##
+      # Activates minitest/ci plugin as a Minitest reporter
+      def report!
+        @report = true
+      end
+
+      ##
+      # Is minitest/ci activated as a Minitest reporter?
+      def report?
+        @report ||= false
+      end
```

Comment:
The only difference in this file are:
- `Minitest.plugin_ci_options`
- `Minitest::Ci.report!`
- `Minitest::Ci.report?`

Otherwise, it's just code from `minitest/ci.rb` moved into `minitest/ci_plugin.rb`